internal/usagegen/usagegen.go

```diff
@@ -115,41 +123,88 @@ func writeFileWithBuffer(buf *bytes.Buffer) func(outFileName string, data []byte
 // writeFormattedDirectory: writes each OperationIDs usage snippet into its own directory with a single main file
 // This will be used frequently by things like devcontainers
 func writeFormattedDirectory(lang, path, content string) error {
-	// Split the input string by the key phrase "Usage snippet provided for"
-	snippets := strings.Split(content, "Usage snippet provided for")
-
-	// Throw out the first line it will just be // or #
-	if strings.Contains(snippets[0], "//") || strings.Contains(snippets[0], "#") {
-		snippets = snippets[1:]
+	snippets, err := ParseUsageOutput(lang, content)
+	if err != nil {
+		return err
 	}
 
 	for _, snippet := range snippets {
-		lines := strings.Split(snippet, "\n")
-
-		// grab the operation name and trim it out of the snippet
-		operationName := strings.TrimSpace(lines[0])
-		lines = lines[1:]
-
-		// remove the trailing line if it includes an empty comment string
-		if strings.TrimSpace(lines[len(lines)-1]) == "//" || strings.TrimSpace(lines[len(lines)-1]) == "#" {
-			lines = lines[0 : len(lines)-1]
-		}
-
-		// trim empty lines at the end of the snippet
-		for len(strings.TrimSpace(lines[len(lines)-1])) == 0 {
-			lines = lines[0 : len(lines)-1]
-		}
+		// TODO: do we still need to do this?
```

Comment:
I can't remember exactly why the empty comment strings were coming up in certain usage snippet generations, but it was causing some formatting issue that was why I was removing it.